src/index.js

```diff
@@ -34,15 +34,24 @@ class LodashModuleReplacementPlugin {
 
   apply(compiler) {
     const resolve = _.memoize(this.resolve, ({ resource }) => resource)
-    compiler.plugin('normal-module-factory', (nmf) => {
-      nmf.plugin('after-resolve', (data, callback) => {
-        if (data) {
-          data.resource = resolve(data)
-          return callback(null, data)
-        }
-        return callback()
+    const afterResolveFn = (data, callback) => {
+      if (data) {
+        data.resource = resolve(data)
+        return callback(null, data)
+      }
+      return callback()
+    }
+
+    /* Webpack >= 4 */
+    if (compiler.hooks) {
+      compiler.hooks.normalModuleFactory.tap('normal-module-factory', (nmf) => {
+        nmf.hooks.afterResolve.tapAsync('after-resolve', afterResolveFn)
```

Comment:
Instead of `normal-module-factory` and `after-resolve` you should pass `LodashModuleReplacementPlugin`. By doing this, the `ProfilingPlugin` will accurately show where the time is spent.

Reply:
In webpack 4, you can use synchronous callbacks for async hooks:

```js
nmf.hooks.afterResolve.tap('LodashModuleReplacementPlugin', data => {
  if (data) {
    data.resource = resolve(data)
  }
  return data
})
```

Reply:
Thanks for the tips :)